unit-tests/test_importgroups.py

```diff
@@ -0,0 +1,137 @@
+            "manager": ["m.manager@yoda.dev"],
+            "member": ["p.member@yoda.dev"],
+            "viewer": ["m.viewer@yoda.dev"],
+        }
+        expected = (
+            "test-automation",
+            "initial",
+            "research-groupteama",
+            ["m.manager@yoda.dev"],
+            ["p.member@yoda.dev"],
+            ["m.viewer@yoda.dev"],
+            "",
+            "",
+        )
+        result, error_msg = _process_csv_line(d, args, "1.9")
+        self.assertTupleEqual(expected, result)
+        self.assertIsNone(error_msg)
+
+        # schema id, expiration date empty strings (should not give an error)
+        d = {
+            "category": ["test-automation"],
+            "subcategory": ["initial"],
+            "groupname": ["groupteama"],
+            "manager": ["m.manager@yoda.dev"],
+            "member": ["p.member@yoda.dev"],
+            "viewer": ["m.viewer@yoda.dev"],
+            "schema_id": [""],
+            "expiration_date": [""],
+        }
+        expected = (
+            "test-automation",
+            "initial",
+            "research-groupteama",
+            ["m.manager@yoda.dev"],
+            ["p.member@yoda.dev"],
+            ["m.viewer@yoda.dev"],
+            "",
+            "",
+        )
+        result, error_msg = _process_csv_line(d, args, "1.9")
+        self.assertTupleEqual(expected, result)
+        self.assertIsNone(error_msg)
+
+        # Missing subcategory
+        d = {
+            "category": ["test-automation"],
+            "groupname": ["groupteama2"],
+            "manager": ["m.manager@yoda.dev"],
+            "member": ["m.member@yoda.dev", "m.member2@yoda.dev"],
+            "expiration_date": ["2030-01-01"],
+            "schema_id": ["default-3"],
+        }
+        result, error_msg = _process_csv_line(d, args, "1.9")
+        self.assertIsNone(result)
+        self.assertIn("missing", error_msg)
+
+    def test_error_fields_1_8(self):
+        args = {"offline_check": True}
+        # Includes (valid) schema id and expiration, even though those are not in version 1.8
+        d = {
+            "category": ["test-automation"],
+            "subcategory": ["initial"],
+            "groupname": ["groupteama"],
+            "manager": ["m.manager@yoda.dev"],
+            "member": ["p.member@yoda.dev"],
+            "viewer": ["m.viewer@yoda.dev"],
+            "schema_id": ["default-3"],
+            "expiration_date": ["2030-01-01"],
+        }
+        result, error_msg = _process_csv_line(d, args, "1.8")
+        self.assertIsNone(result)
+        self.assertIn("1.9", error_msg)
+
+    def test_parse_invalid_csv_file(self):
+        # csv that has an unlabeled header
+        args = {"offline_check": True}
+        with self.assertRaises(SystemExit):
+            parse_csv_file("files/unlabeled-column.csv", args, "1.9")
+
+        # csv that has too many items in the rows compared to the headers
+        with self.assertRaises(SystemExit):
```

Comment:
When these are run, the errors are printed to the screen, which might be confusing for a user, but I'm not sure if there is a way to suppress them.

Reply:
I think it's possible to suppress these error messages by either changing the test runner or by mocking `stderr` for the  test that results in the output being printed. Since the output is shown for only one test, I think just mocking `stderr` there is practical.

Something along the lines of:

```python
from io import StringIO
from unittest.mock import patch

# other code here

    @patch('sys.stderr', new_callable=StringIO)
    def test_parse_invalid_csv_file(self, mock_stderr):
        # Test implementation here
```